src/sparse/sparse_matrix.h

```diff
@@ -0,0 +1,131 @@
+/*!
+ *  Copyright (c) 2022 by Contributors
+ * \file sparse/sparse_matrix.h
+ * \brief DGL C++ sparse matrix header
+ */
+#ifndef DGL_SPARSE_SPARSE_MATRIX_H_
+#define DGL_SPARSE_SPARSE_MATRIX_H_
+
+#include <torch/custom_class.h>
+#include <torch/script.h>
+
+#include <memory>
+
+namespace dgl {
+namespace sparse {
+
+/*! \brief SparseFormat enumeration */
+enum SparseFormat { kCOO, kCSR, kCSC };
+
+/*! \brief CSR sparse structure */
+struct CSR {
+  torch::Tensor indptr, indices;
+  // The element order of the sparse format. In the SparseMatrix, we have data
+  // (value_) for each non-zero values. The elements in sparse formats may not
+  // follow the same order as in the SparseMatrix. So we store `e_order` in a
```

Comment:
I cannot parse the sentence "The elements in sparse formats may not follow the same order as in the SparseMatrix."

How about

> The order of non-zero values in (value_) may not follow the order of non-zero entries in the matrix.